cmd/exporter.go

```diff
@@ -42,6 +42,13 @@ var (
 	enabledEBPFCgroupID = flag.Bool("enable-cgroup-id", true, "whether enable eBPF to collect cgroup id (must have kernel version >= 4.18 and cGroup v2)")
 )
 
+func healthProbe(w http.ResponseWriter, req *http.Request) {
```

Comment:
@rootfs @marceloamaral what do you think best way to probe exporter? check with the prometheus endpoint? Thanks

Reply:
healthz looks good to me. If the collector is not working, the metrics won't be there. 
Can you add a component level health status report and embed them in the HTTP response? In this way, we know the health status of attacher, collector, etc

Reply:
The initial PR can get the framework ready, followed by per component level health message construction

Reply:
So, is the idea to create a liveness probe configuration later?
https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/

Reply:
yes, I am looking for a way to know how to detect it's a health pod 
e.g something is running so we can say it's healthy or not ..

the healthy setup has been done in the manifests already but not sure what to check now

Reply:
Just check if the http server replies anything, so it means that the container is running

Reply:
@rootfs are you looking for something like following?
/healthz

status 200 
module1 : ok
module2 : fail
module3 : ok

Reply:
can we use any additional http status code? https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

e.g. if attacher is up, but collector fail, return 201. If some but not all metrics are available, return 206. All the codes are documented for diagnosis.

Reply:
um... 

health works like following way

https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
```
Any code greater than or equal to 200 and less than 400 indicates success. Any other code indicates failure.
```
So I think return 201 or 206 make no difference 


and from other projects I worked before 
https://github.com/kubernetes-csi/livenessprobe/blob/master/cmd/livenessprobe/main.go#L70
they have a sidecar (but seems kepler no need) and it check some other pods running well or through rpc
looks like we don't have such need as well

so guess the simplest way is to always return ok check here 
if there is something wrong in access http:<host>:9102/healthz , that means pod is not healthy (since <host>:9102/metrics will not work at all)

thoughts?

Reply:
ok, that makes more sense. It returns 4xx when the metrics collector fail, otherwise just return 200.